scripts/generate_forecasts.py

```diff
@@ -0,0 +1,91 @@
+    DataFrames into a single DataFrame and returns it.
+
+    Args:
+        sites_info (list): List of tuples containing site information. Each tuple should be in the format
+            (pv_id, latitude, longitude, capacity). Latitude and longitude are geographic coordinates, and capacity
+            is the site's capacity in kilowatts peak (kWp).
+        forecast_date (str): Date for which the forecast is generated (format: "YYYY-MM-DD").
+
+    Returns:
+        pandas.DataFrame: DataFrame containing forecasts for each PV site. The DataFrame has columns for each site's
+            latitude, longitude, capacity, and power forecast, with a column name in the format "{pv_id} Power".
+            The index of the DataFrame is set to the forecast dates.
+    """
+    all_forecasts = []  # List to store DataFrames for each site
+
+    # Loop through each site information
+    for site_info in sites_info:
+        # Unpack site information from the tuple
+        pv_id, latitude, longitude, capacity = site_info
+
+        # Create PVSite object for the site
+        site = PVSite(latitude=latitude, longitude=longitude, capacity_kwp=capacity)
+
+        # Run forecast for the site
+        forecast = run_forecast(site=site, ts=forecast_date)
+
+        # Flatten forecast values to a 1D array
+        forecast_values = forecast.values.flatten()
+
+        # Extract forecast dates from the forecast index
+        forecast_dates = forecast.index
+
+        # Create a DataFrame for the current site
+        site_df = pd.DataFrame(
+            {
+                f"{pv_id}_latitude": [latitude]
+                * len(forecast_dates),  # Repeat latitude for each forecast date
+                f"{pv_id}_longitude": [longitude]
+                * len(forecast_dates),  # Repeat longitude for each forecast date
+                f"{pv_id}_capacity": [capacity]
+                * len(forecast_dates),  # Repeat capacity for each forecast date
+                f"{pv_id} Power": forecast_values,  # Power forecast column with renamed header
+            },
+            index=forecast_dates,
+        )
+
+        # Append the site DataFrame to the list
+        all_forecasts.append(site_df)
+
+    # Concatenate all site DataFrames into a single DataFrame along the column axis
+    master_df = pd.concat(all_forecasts, axis=1)
+
+    return master_df
+
+
+if __name__ == "__main__":
+    # Example input for multiple PV sites
+    sites_info = [
+        (
+            "Site1",
+            51.75,
+            -1.25,
+            1.25,
+        ),  # Site 1 information (pv_id, latitude, longitude, capacity)
+        (
+            "Site2",
+            52.0,
+            -1.5,
+            1.5,
+        ),  # Site 2 information (pv_id, latitude, longitude, capacity)
+        # Add more sites as needed, with format (pv_id, latitude, longitude, capacity)
+    ]
+    forecast_date = "2023-11-01"  # Forecast date
+    output_file = "pv_forecasts.csv"  # Output file name
```

Comment:
Could you change the output file name to include the forecast_date, and also mention that is for multiple sites. Maybe "multiple_pv_forecasts_{forecast_date}"?

Reply:
Yeah will do. Have had my finals this week. Will have it pushed by tonight

Reply:
Ok. Have had my finals this week. Will have the changes pushed by tonight